.github/workflows/tests.yaml

```diff
@@ -0,0 +1,29 @@
+name: Unit Tests
+on: [push, pull_request_target]
+
+jobs:
+  tests:
+    name: Tests (Go ${{ matrix.go-version }})
+    runs-on: ubuntu-latest
+    # always run on push events
+    # only run on pull_request_target event when pull request pulls from fork repository
+    if: >
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.full_name != github.repository 
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.11, 1.12, 1.13, 1.14, 1.15, "^1"]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - run: ./sh/gofmt.sh
```

Comment:
Are we sure that this will fail the action if it's not formatted correctly? Doesn't seem to give us much info. in the output on this action when it's run. Same with `go vet`

Reply:
AFAIK both tools return a shell exit code of 1 if they fail, which the run step will check for.